modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -47,7 +52,29 @@ func GeneratePortID(owner, connectionID, counterpartyConnectionID string) (strin
 		return "", sdkerrors.Wrap(err, "invalid counterparty connection identifier")
 	}
 
-	return fmt.Sprintf("%s-%d-%d-%s", VersionPrefix, connectionSeq, counterpartyConnectionSeq, owner), nil
+	return fmt.Sprint(VersionPrefix, Delimiter, connectionSeq, Delimiter, counterpartyConnectionSeq, Delimiter, owner), nil
+}
+
+// ParseControllerConnSequence attempts to parse the controller connection sequence from the provided port identifier
+// The port identifier must match the controller chain format outlined in (TODO: link spec), otherwise an empty string is returned
+func ParseControllerConnSequence(portID string) string {
```

Comment:
My preference would be to return a uint64. Is there a reason for returning a string instead? I guess then you'd need to return an error

Reply:
I agree, uint64 would be more appropriate here. I originally opted for the `strconv` calls in `validateConnectionParams` to avoid having to do another error check, but it would look more fitting to return uint64! It will just add additional error checking to the handshake handler.